### ISSUE TITLE
Pendulum-1, MountainCarContinuous-v0 and Reacher-misc return non-squeezed reward

### ISSUE THREAD

**Issue by keraJLi** [RobertTLange/gymnax#43]
Out of all environments, Pendulum-1, MountainCarContinuous-v0 and Reacher-misc return a jax array with shape `(1, )` as a reward. This is inconsistent with all other environments, which return an array with shape `()`. This can lead to unexpected shaping errors, for example consider a case like this
```
num_envs = 3
weights = jnp.arange(num_envs)
rewards, ... = jax.vmap(env.step, ...)(actions)
weighted_rewards = weights * rewards
```
If the reward returned by the environment has shape `(1, )`, the the result of vmapping will have shape `(3, 1)` instead of `(3, )`, and therefore `weighted_rewards` will have shape `(3, 3)` instead of `(3, )`.
